### PR TITLE
Copy functions from `ServiceHubContextBuilder`  to `ServiceManagerBuilder`

### PR DESCRIPTION
**Pull request by Y-Sindo**
Prepare to merge two builder into one to reduce user migration cost.

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs [hunk 1]
```diff
@@ -84,16 +100,56 @@ public ServiceManagerBuilder WithCallingAssembly()
...
         /// Builds <see cref="IServiceManager"/> instances.
         /// </summary>
         /// <returns>The instance of the <see cref="IServiceManager"/>.</returns>
         public IServiceManager Build()
         {
-            return _services.AddSignalRServiceManager()
-                .AddSingleton(_services.ToList() as IReadOnlyCollection<ServiceDescriptor>)
-                .BuildServiceProvider()
+            var serviceCollection = new ServiceCollection().Add(_services)
+                .AddSingleton(_services.ToList() as IReadOnlyCollection<ServiceDescriptor>);
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs:116]: `services` is added twice?

File: src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs [hunk 2]
```diff
@@ -84,16 +100,56 @@ public ServiceManagerBuilder WithCallingAssembly()
...
+            _configureAction?.Invoke(serviceCollection);
+            return serviceCollection.BuildServiceProvider()
                 .GetRequiredService<IServiceManager>();
         }
+
+        /// <summary>
+        /// Builds <see cref="ServiceHubContext"/> instances.
+        /// </summary>
+        /// <returns>The instance of the <see cref="IServiceManager"/>.</returns>
+        public async Task<ServiceHubContext> CreateHubContextAsync(string hubName, CancellationToken cancellationToken)
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs:126]: `public` or `internal` first?

File: src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs [hunk 3]
```diff
@@ -84,16 +99,56 @@ public ServiceManagerBuilder WithCallingAssembly()
...
+            //initialize
+            var connectionContainer = serviceHubContext.ServiceProvider.GetService<IServiceConnectionContainer>();
+            if (connectionContainer != null)
+            {
+                await connectionContainer.ConnectionInitializedTask.OrTimeout(cancellationToken);
+            }
+
+            if (cancellationToken.IsCancellationRequested)
+            {
+                await serviceHubContext?.DisposeAsync();
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs:148]: ConnectionInitializedTask can throw, wrap this into finally?

Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs:150]: I think we should dispose only if exception thrown, so wrap them into catch.

Comment (vicancy) [reply src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs:150]: do you want to also throw when `cancellationToken.IsCancellationRequested`

**Review by vicancy — APPROVED** [review]

**Pull request merged**